hal/src/sercom/v2/i2c.rs

```diff
@@ -0,0 +1,137 @@
+//! Use the SERCOM peripheral for I2C communications.
+
+mod pads;
+pub use pads::*;
+
+mod reg;
+use reg::Registers;
+
+mod flags;
+pub use flags::*;
+
+mod config;
+pub use config::*;
+
+pub mod impl_ehal;
```

Comment:
If you make this module `pub`, then I would add documentation there on the various `embedded-hal` traits and their implementations. Otherwise, make it private.